File: PerfTest1/PerfTest1-Java/app/src/main/java/com/vandammeford/kevinf/perftest1_java/CalcPrimes.java
package com.vandammeford.kevinf.perftest1_java;

import android.os.Bundle;
import android.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;


public class CalcPrimes extends Fragment {

    public CalcPrimes() {
        // Required empty public constructor
    }

    @Override
    public void onStart() {
        super.onStart();
        Button btnCalcPrimes = (Button)this.getActivity().findViewById(R.id.btnCalcPrimes);

        btnCalcPrimes.setOnClickListener((View.OnClickListener)this.getActivity());
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_calc_primes, container, false);
    }

    @Override
    public void onDetach() {
        super.onDetach();
        Button btnCalcPrimes = (Button)this.getActivity().findViewById(R.id.btnCalcPrimes);

        if (btnCalcPrimes != null) {
            btnCalcPrimes.setOnClickListener(null);
        }
    }
}

File: PerfTest1/PerfTest1-Xamarin/PerfTest1-ClassicXamarin.Android/CalcPrimes.cs
using Android.App;
using Android.OS;
using Android.Views;
using Android.Widget;

namespace PerfTest1Xamarin
{
    public class CalcPrimes : Fragment
    {

        public CalcPrimes()
        {
            // Required empty public constructor
        }

        public override void OnStart()
        {
            base.OnStart();
            Button btnCalcPrimes = (Button) this.Activity.FindViewById(Resource.Id.btnCalcPrimes);

            btnCalcPrimes.SetOnClickListener((View.IOnClickListener) this.Activity);
        }

        public override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
        }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState)
        {
            // Inflate the layout for this fragment
            return inflater.Inflate(Resource.Layout.fragment_calc_primes, container, false);
        }

        public override void OnDetach()
        {
            base.OnDetach();
            Button btnCalcPrimes = (Button) this.Activity.FindViewById(Resource.Id.btnCalcPrimes);

            if (btnCalcPrimes != null)
            {
                btnCalcPrimes.SetOnClickListener(null);
            }
        }
    }
}